### DISCUSSION TITLE
Maybe I can contribute

### DISCUSSION THREAD

**Discussion by Boy19991** [FreeTubeApp/FreeTube#2368] [Q&A] [answered]
Ive noticed freetube is not functioning well in the last few days.

I read https://blog.freetubeapp.io/the-invidious-project-will-be-shutting-down-soon

Invidious API error and various other erros are popping up and it has become unusable.

I have novice/intermediate PHP/Javascript knowledge, maybe I could help out if someone could give me some guidance.

Is recieving some guidance a possibility?

Nick.

**Comment by PrestonN** [comment, marked as answer]
Hey there!

We're always looking for individuals who would like to contribute to FreeTube and I'm glad that you're interested in doing so! Feel free to take a look at our [contribution guidliines](https://github.com/FreeTubeApp/FreeTube/blob/development/CONTRIBUTING.md) article for information on getting started.

That article you linked is fairly old at this point, and both FreeTube and Invidious are very different since I wrote it. At that time, FreeTube was using a different code base and the future of Invidious was unknown. Today FreeTube has a much nicer code base and Invidious has developed a team of individuals to support the project and they continue to do so to this day.

If you believe that there are improvements that you can make to the Invidious API, I'm sure the team on the [Invidious Project](https://github.com/iv-org/invidious) would be more than happy to have help over there as well.

For FreeTube, we primarily use JavaScript / Vue.js so knowledge with those is preferred. If you'd specifically like to contribute to our local API, check out [this article](https://docs.freetubeapp.io/usage/local-api/#modules-used-in-the-local-api) as it lays out the individual projects that make up our local API. Improvements towards that specific logic will need to go upstream to those projects. Some of them we do not own so you'll need to be aware of separate guidelines that those projects may have.

If you'd like to try your hand at something else, there are quite a few open issues that need to be looked at as you can probably tell. Feel free to drop a comment on that ticket that you'd like to work on and we'll assign you to it. Once you're done, feel free to create a PR and our team will review it.

We also currently have a release candidate in progress. Feel free to check out more information [here](https://github.com/FreeTubeApp/FreeTube/discussions/2309) and test out the new features before we go live. Testing is also very helpful.

If you ever have any questions about the code, we're all fairly active over on our [Matrix Community](https://matrix.to/#/+freetube:matrix.org) so feel free to hop in and chat with us and a number of others.

Thanks again for your interest, hope to see a contribution in the future!
